pyproject.toml

```diff
@@ -38,7 +38,7 @@ dependencies = [
     "numpy>=1.16.0,<2.0.0",
     "scipy>=1.4.0,<2.0.0",
     "h5py>=3.1.0,<4.0.0",
-    "opencv-python>=4.5.0,<5.0.0",
+    "opencv-python-headless>=4.5.0,<5.0.0",
```

Comment:
This change would require a little more thought. Here, it means that we can no longer use the view methods otherwise it will generate an error for users and they will have to reinstall/overwrite the current installation with `opencv-python`.

I'm more in favour of keeping the non-headless package for now and find a way to have headless by default and have an optional dependency so it's not `opencv-python-headless` but `opencv-python`.
It's harder than having just an optional dependancy: it should replace `opencv-python-headless` by the other one and I'm not sure we can do that with `pyproject.toml`.

Reply:
> This change would require a little more thought. Here, it means that we can no longer use the view methods otherwise it will generate an error for users and they will have to reinstall/overwrite the current installation with `opencv-python`.
> 
> I'm more in favour of keeping the non-headless package for now and find a way to have headless by default and have an optional dependency so it's not `opencv-python-headless` but `opencv-python`. It's harder than having just an optional dependancy: it should replace `opencv-python-headless` by the other one and I'm not sure we can do that with `pyproject.toml`.

you mean the visualization we use at doctr ? That's done with matplotlib so doctr itself doesn't requires opencv GUI support :)

Reply:
ok tested and the behaviour is strange so yeah you are right would require something like `python-doctr[headless]`

Switch PR to only CI updates :)